CMakeLists.txt

```diff
@@ -1,23 +1,31 @@
 # Copyright 2011-2018 Sebastian A. Mueller
 cmake_minimum_required(VERSION 2.8)
-set (CMAKE_CXX_FLAGS "-std=c++1z -pthread -Wall -Wextra -pedantic -O1 -g -I")
+
 project(merlict)
 
-# Assert compiler supports C++ 11
+# Assert compiler supports C++ 17
 include(CheckCXXCompilerFlag)
-CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
-if(COMPILER_SUPPORTS_CXX11)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+CHECK_CXX_COMPILER_FLAG("-std=c++17" COMPILER_SUPPORTS_CXX17)
+if(COMPILER_SUPPORTS_CXX17)
+    set(CMAKE_CXX_FLAGS "-std=c++17 -pthread -Wall -Wextra -pedantic -I")
 else()
     message(
         STATUS
-        "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support.
+        "The compiler ${CMAKE_CXX_COMPILER} has no C++17 support.
         Please use a different C++ compiler.")
 endif()
 
+if(USE_GCOV)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -O0 --coverage")
+    message(STATUS "The compiler flags ${CMAKE_CXX_FLAGS} are set for coverage reports.")
+else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -march=native")
```

Comment:
Is there a specific reason for g++ optimization-flag '-O3'? I would like to keep '-O1' instead as it is considered to be more compatible. Right now '-O3' gives even 3.5 times slower execution than the master with '-O1'.
The problem is, when I change this '-O3' to '-O1', it has no effect. But when I put the '-O1' flag again into line 10, where CMAKE_CXX_FLAGS is first set the execution is fast again.